src/message_sender.c

```diff
@@ -562,7 +562,6 @@ static void on_link_state_changed(void* context, LINK_STATE new_link_state, LINK
         {
             /* User initiated transition, we should be good */
             set_message_sender_state(message_sender_instance, MESSAGE_SENDER_STATE_IDLE);
-            indicate_all_messages_as_error(message_sender_instance);
```

Comment:
Hi, I can check if this solves my issue as well and update with a new commit afterwards.